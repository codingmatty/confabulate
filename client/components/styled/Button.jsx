import styled from 'styled-components';

function colorByType({ theme, type }) {
  switch (type) {
    case 'primary':
      return `
      background-color: ${theme.color.blues[2]};
      border: 1px solid ${theme.color.blues[7]};
      `;
    case 'destructive':
      return `
      background-color: ${theme.color.reds[4]};
      border: 1px solid ${theme.color.reds[7]};
      color: white;
      `;
    case 'transparent':
      return `
      background-color: transparent;
      border: 1px solid transparent;
      `;
    case 'neutral':
      return `
      background-color: ${theme.color.neutrals[1]};
      border: 1px solid ${theme.color.neutrals[7]};
      `;
    default:
      return `
      background: none;
      border: 1px solid ${theme.color.neutrals[5]};
    `;
  }
}

export default styled.button`
  ${colorByType};
  border-radius: 2px;
  font-size: 1rem;
  padding: 0.5rem 2rem;

  ${({ fullWidth }) => fullWidth && `width: 100%`};
`;
