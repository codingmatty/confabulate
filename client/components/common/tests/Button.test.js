import { cleanup } from 'react-testing-library';
import renderWithTheme from '../../../test-utils/render-with-theme';
import Button from '../Button';

describe('Button', () => {
  afterEach(cleanup);

  it('matches default snapshot', () => {
    const { container } = renderWithTheme(<Button />);
    expect(container).toMatchSnapshot();
  });

  it('applies primary styles', () => {
    const { container } = renderWithTheme(<Button type="primary" />);
    expect(container).toMatchSnapshot();
  });

  it('applies destructive styles', () => {
    const { container } = renderWithTheme(<Button type="destructive" />);
    expect(container).toMatchSnapshot();
  });

  it('applies transparent styles', () => {
    const { container } = renderWithTheme(<Button type="transparent" />);
    expect(container).toMatchSnapshot();
  });

  it('applies neutral styles', () => {
    const { container } = renderWithTheme(<Button type="neutral" />);
    expect(container).toMatchSnapshot();
  });

  it('adds full width style', () => {
    const { container } = renderWithTheme(<Button fullWidth />);
    expect(container.firstChild).toHaveStyleRule('width', '100%');
  });
});
