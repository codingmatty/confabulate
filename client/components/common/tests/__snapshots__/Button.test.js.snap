// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button applies destructive styles 1`] = `
.c0 {
  background-color: #D64545;
  border: 1px solid #911111;
  color: white;
  border-radius: 2px;
  font-size: 1rem;
  padding: 0.5rem 2rem;
}

<div>
  <button
    class="c0"
    type="destructive"
  />
</div>
`;

exports[`Button applies neutral styles 1`] = `
.c0 {
  background-color: #E1E1E1;
  border: 1px solid #515151;
  border-radius: 2px;
  font-size: 1rem;
  padding: 0.5rem 2rem;
}

<div>
  <button
    class="c0"
    type="neutral"
  />
</div>
`;

exports[`Button applies primary styles 1`] = `
.c0 {
  background-color: #84C5F4;
  border: 1px solid #0F609B;
  border-radius: 2px;
  font-size: 1rem;
  padding: 0.5rem 2rem;
}

<div>
  <button
    class="c0"
    type="primary"
  />
</div>
`;

exports[`Button applies transparent styles 1`] = `
.c0 {
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 2px;
  font-size: 1rem;
  padding: 0.5rem 2rem;
}

<div>
  <button
    class="c0"
    type="transparent"
  />
</div>
`;

exports[`Button matches default snapshot 1`] = `
.c0 {
  background: none;
  border: 1px solid #7E7E7E;
  border-radius: 2px;
  font-size: 1rem;
  padding: 0.5rem 2rem;
}

<div>
  <button
    class="c0"
  />
</div>
`;
